drivers/hud_spm_report/app/models/hud_spm_report/fy2024/episode.rb

```diff
@@ -107,12 +110,18 @@ def compute_episode(enrollments, included_project_types:, excluded_project_types
         )
       end
 
-      dates = filtered_bed_nights.map(&:last)
+      first_date, last_date, days_homeless = compute_min_max_and_unique(filtered_bed_nights.map(&:last))
```

Comment:
several issues here:
* filtered_bed_nights were not sorted as seems to have been the assumption. This led to inaccurate start/end dates for episodes
* the dates themselves were not de-duplicated before counting to find the days_homeless which led to inaccurate durations